internal/controller/pkg/manager/reconciler.go

```diff
@@ -390,9 +390,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		}
 	}
 
-	if pr.GetCondition(v1.TypeHealthy).Status == corev1.ConditionTrue {
-		p.SetConditions(v1.Healthy())
+	// TODO(phisco): refactor these conditions to make it clearer
+	if pr.GetCondition(v1.TypeHealthy).Status == corev1.ConditionTrue && p.GetCondition(v1.TypeHealthy).Status != corev1.ConditionTrue {
+		// NOTE(phisco): We don't want to spam the user with events if the
+		// package is already healthy, nor update the condition last transition
+		// time if it didn't actually change.
```

Comment:
Hrm, I would not have thought we needed to do this. SetConditions should already be a no-op if we're setting a condition that already existed. 🤔

Reply:
🤔 you are right, I guess didn't get deep enough in the involved interfaces 😩 let me remove the comment and move back the condition setting out of the if

Reply:
Pushed